pkg/engine/filetransfer.go

```diff
@@ -56,26 +87,68 @@ func fileTransferPodman(ctx context.Context, path, dest, method string, target *
 	s.Command = []string{"sh", "-c", "cp " + copyFile}
 	s.Mounts = []specs.Mount{{Source: dest, Destination: dest, Type: "bind", Options: []string{"rw"}}}
 	s.Volumes = []*specgen.NamedVolume{{Name: harpoonVolume, Dest: "/opt", Options: []string{"ro"}}}
-	createResponse, err := containers.CreateWithSpec(conn, s, nil)
+	return s
+}
+
+func generateSpecRemove(method, file, pathToRemove, dest string, target *api.Target) *specgen.SpecGenerator {
+	s := specgen.NewSpecGenerator(harpoonImage, false)
+	s.Name = method + "-" + target.Name + "-" + file
+	s.Privileged = true
+	s.PidNS = specgen.Namespace{
+		NSMode: "host",
+		Value:  "",
+	}
+	s.Command = []string{"sh", "-c", "rm " + pathToRemove}
+	s.Mounts = []specs.Mount{{Source: dest, Destination: dest, Type: "bind", Options: []string{"rw"}}}
+	s.Volumes = []*specgen.NamedVolume{{Name: harpoonVolume, Dest: "/opt", Options: []string{"ro"}}}
+	return s
+}
+
+func fetchImage(conn context.Context) error {
```

Comment:
I think it's better to name all go context variables as `ctx`. `conn` is easily confused to be a connection. k8s uses `ctx` almost a lot!